src/app/modules/core/services/events.service.ts

```diff
@@ -0,0 +1,12 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, Event, NavigationEnd } from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class EventsService {
+  routeChanged$ = new BehaviorSubject<ActivatedRouteSnapshot>({} as ActivatedRouteSnapshot);
```

Comment:
A behavior subject is great because subscribers will always receive it the first time they subscribe to it